### PR TITLE
Dates and times should follow GDS style guide

### PR DESCRIPTION
**Pull request by paulrobertlloyd**
[Dates](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#dates) and [times](https://www.gov.uk/guidance/style-guide/a-to-z-of-gov-uk-style#times) should follow the GDS style guide.

This PR adds a date formatter that can be called with `.to_formatted_s(:govuk_date)` and `.to_formatted_s(:govuk_time)`, amongst other common options.

### CODE REVIEW

File: spec/models/form/question_spec.rb [hunk 1]
```diff
@@ -118,7 +118,7 @@
 
       it "displays a formatted answer label" do
         case_log.mrcdate = Time.zone.local(2021, 10, 11)
-        expect(subject.answer_label(case_log)).to eq("11 Oct 2021")
+        expect(subject.answer_label(case_log)).to eq("11 October 2021")
```
Comment (baarkerlounger) [inline spec/models/form/question_spec.rb:121]: Would be good to have tests for Time display as well (particularly since there's a couple of edge cases there (Midnight, Midday ect)

File: config/initializers/date_formats.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+Time::DATE_FORMATS[:month_and_year] = "%B %Y"
+Date::DATE_FORMATS[:month_and_year] = "%B %Y"
+
+Time::DATE_FORMATS[:short_month_and_year] = "%b %Y"
+Date::DATE_FORMATS[:short_month_and_year] = "%b %Y"
+
+Time::DATE_FORMATS[:day_and_month] = "%-d %B"
+Date::DATE_FORMATS[:day_and_month] = "%-d %B"
+
+Time::DATE_FORMATS[:govuk_date_and_time] = lambda do |time|
```
Comment (baarkerlounger) [inline config/initializers/date_formats.rb:15]: What's the benefit of putting this in an initializer rather than a module?

Comment (paulrobertlloyd) [reply config/initializers/date_formats.rb:15]: 🤷🏼‍♂️ I just copied this from https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/config/initializers/date_formats.rb – BUSTED! I’ve no idea what might be considered best practice, but it works. 😊

Comment (baarkerlounger) [reply config/initializers/date_formats.rb:15]: Would recommend copy and pasting all of it then :stuck_out_tongue: 

https://github.com/DFE-Digital/apply-for-teacher-training/blob/main/spec/config/initializers/date_formats_spec.rb

Comment (baarkerlounger) [reply config/initializers/date_formats.rb:15]: For reference the downside of this is changes to this code will require a server restart to see in dev. Not super important.

Comment (paulrobertlloyd) [reply config/initializers/date_formats.rb:15]: God damn it, I looked for a spec, but obviously not hard enough 😄

File: spec/models/form/question_spec.rb (hunk 1 above)
Comment (paulrobertlloyd) [reply spec/models/form/question_spec.rb:121]: Added initializer spec.

**Review by baarkerlounger — APPROVED** [review]
